### PR TITLE
Error message component

### PR DESCRIPTION
**Pull request by tmansuroglu**
This separated from #99

### CODE REVIEW

**Review by louisrli — COMMENTED** [review]
proofread

**Comment by tmansuroglu** [comment]
@louisrli extra snapshots are cleared from PR. PTAL

File: src/components/userauth/ErrorMessage/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`ErrorMessage snapshot test 1`] = `null`;
```
Comment (louisrli) [inline src/components/userauth/ErrorMessage/__snapshots__/index.test.js.snap]: this test just says null.

add a second test where you pass an error message please

Comment (tmansuroglu) [reply src/components/userauth/ErrorMessage/__snapshots__/index.test.js.snap]: I didn't understand what you meant by second test but I passed a test message as prop to the test component. Now snapshot is not null

Comment (louisrli) [reply src/components/userauth/ErrorMessage/__snapshots__/index.test.js.snap]: you are allowed to have more than one test in a file. and you should most of the time, especially if there are two cases

File: src/components/userauth/ErrorMessage/index.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import ErrorMessage from ".";
+
```
Comment (louisrli) [inline src/components/userauth/ErrorMessage/index.test.js:4]: "it should return null if there is no error"

File: src/components/userauth/ErrorMessage/__snapshots__/index.test.js.snap (hunk 1 above)
Comment (tmansuroglu) [reply src/components/userauth/ErrorMessage/__snapshots__/index.test.js.snap]: So I should be displaying all conditions inside snapshot. I get it.

File: src/components/userauth/ErrorMessage/index.test.js (hunk 2 above)
Comment (tmansuroglu) [reply src/components/userauth/ErrorMessage/index.test.js:4]: Returning null doesn't mean something is wrong. Message only displayed when user makes mistake. Most of the time there shouldn't be a message.

Comment (tmansuroglu) [reply src/components/userauth/ErrorMessage/index.test.js:4]: I feel like im missing your point.

Comment (louisrli) [reply src/components/userauth/ErrorMessage/index.test.js:4]: your code is correct but i'm aware that returning null doesnt mean something is wrong and that its intended -- thats why we write a test case to express that intention

**Review by louisrli — APPROVED** [review]

File: src/components/userauth/ErrorMessage/index.test.js (hunk 2 above)
Comment (louisrli) [reply src/components/userauth/ErrorMessage/index.test.js:4]: but in general if you have only 1 test and it all it returns is null its useless for a snapshot. so in this case we definitely want the non-null case covered.

Comment (louisrli) [reply src/components/userauth/ErrorMessage/index.test.js:4]: for the null case we wouldnt normally use a snapshot test to be honest. we'd just use jest expect(foo).toBeNull

Comment (ssinikoglu) [reply src/components/userauth/ErrorMessage/index.test.js:4]: I'm also not sure why exactly it is, but null type is defined as object ( I've just found it via MDN [link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null) and in the code, message is defined as string. This difference may cause a problem.

Comment (tmansuroglu) [reply src/components/userauth/ErrorMessage/index.test.js:4]: hmm I see. Thank you Louis

**Review by ssinikoglu — APPROVED** [review]

**Comment by tmansuroglu** [comment]
@Ammar-64 can you merge?

**Pull request merged**